source/common/ur_util.hpp

```diff
@@ -264,6 +264,10 @@ inline std::size_t combine_hashes(std::size_t seed, const T &v, Args... args) {
     return combine_hashes(seed ^ std::hash<T>{}(v), args...);
 }
 
+struct ur_exception_t {
```

Comment:
what's the reason for wrapping this? Can't we just throw the ur_result_t, like it happens in other places?

Reply:
I was thinking about introducing additional struct and potentially inheriting from std::exception but I don't think we need it for now (it could be useful if we were to do some logging in catch()). I removed this.